pymongo/database.py

```diff
@@ -541,36 +541,39 @@ def command(self, command, value=1, check=True,
                                  check, allowable_errors, read_preference,
                                  codec_options, session=session, **kwargs)
 
-    def _list_collections(self, sock_info, slave_okay, criteria=None,
-                          session=None):
+    def list_collections(self, sock_info, slave_okay, filter=None,
+                         session=None, batch_size=0):
         """Internal listCollections helper."""
-        criteria = criteria or {}
-        cmd = SON([("listCollections", 1), ("cursor", {})])
-        if criteria:
-            cmd["filter"] = criteria
+        filter = filter or {}
+        cmd = SON([("listCollections", 1),
+                   ("cursor", {"batchSize": batch_size}),
+                   ("filter",filter)])
```

Comment:
`filter` should only be added if it's non-empty and `batch_size` only if it's not `None`:
```python
cmd = SON([("listCollections", 1), ("cursor", {})])
if filter:
    cmd["filter"] = filter
if batch_size is not None:
    cmd["cursor"]["batchSize"] = batch_size
```